tests/ui/fmt/suggest-wrongly-order-format-parameter.rs

```diff
@@ -0,0 +1,25 @@
+//! Regression test for https://github.com/rust-lang/rust/issues/129966
+//!
+//! Ensure we provide suggestion for wrongly ordered format parameters.
+
+//@ run-rustfix
+#![allow(dead_code)]
+
+#[derive(Debug)]
+struct Foo(u8, u8);
+
+fn main() {
+    let f = Foo(1, 2);
+
+    println!("{f:?#}");
+    //~^ ERROR invalid format string: expected `}`, found `#`
+    //~| HELP did you mean `#?`?
+
+    println!("{f:?x}");
+    //~^ ERROR invalid format string: expected `}`, found `x`
+    //~| HELP did you mean `x?`?
```

Comment:
I thought it made sense to return the same error description, `expected '}'`..., rather than introducing a new one, as it fits  with the spotted error. I added a suggestion to reorder the format parameter if a `#/x/X` character is spotted afterward. However, I completely understand if the preference is to add a new error message similar to the approach taken in the previous PR this one replaces.